content/guides/kubernetes/ytt-gs.md

```diff
@@ -0,0 +1,269 @@
+https://carvel.dev/ytt/#example:example-plain-yaml 
+
+Using the `--output pos` flag with the CLI is also a great debugging tool -- It shows you where each change comes from!
+You can also choose to write out to separate YAML files using the `--output-files` flag.
+
+
+## Directives and Annotations
+ytt uses YAML comments for ytt annotations and ytt directives.
+Annotations, such as `#@data/values`, do not have a space and attach metadata to a YAML node.
+
+ytt directives, on the other hand, have a space such as `#@ if` and `#@ load` and are used to direct ytt to execute the arguments after it. 
+
+Additionally, instead of using just `#` for standard comments, they should be `#!` to avoid “linting” errors.
+
+## Types and Control Flow
+Within the comments, ytt uses a slightly modified version of the [Starlark](https://github.com/google/starlark-go/blob/master/doc/spec.md) programming language which is a dialect of Python.
+
+When we need to process data, declaring data using ytt directives instead of plain YAML nodes can be helpful. ytt will convert these values to their respective YAML types.
+For a quick example of all of the types and how they output to YAML, see [this example](https://carvel.dev/ytt/#example:example-datatypes) on the ytt playground.
+
+The [ytt language reference](https://carvel.dev/ytt/docs/latest/lang/) shows the standard methods you can use on strings and collections. And it details the if statement, for-loop, and function control-flow structures.
+
+One example of using an if statement is as follows:
+```
+#@ if True:
+test1: 123
+#@ else:
+test2: 124
+#@ end
+```
+
+
+And here is a for loop:
+```
+array:
+#@ for i in range(0,3):
+- #@ i
+- #@ i+1
+#@ end
+```
+## Variables, Functions, and Fragments
+In this example variables are used to store the app name and version in a single place.
+A function is then used to define a repeatable set of labels using a [YAML Fragment](https://carvel.dev/ytt/docs/latest/lang-ref-yaml-fragment/) with an optional parameter to include the version number.
+
+Variable definition:
+```yaml
+#@ app_name = "prometheus-operator"
+#@ version = "v0.39.0"
+```
+
+Function with YAML Fragment in it:
+```yaml
+#@ def labels(with_version=False):
+  app.kubernetes.io/component: controller
+  app.kubernetes.io/name: #@ app_name
+  #@ if with_version:
+  app.kubernetes.io/version: #@ version
+  #@ end
+#@ end
+```
+
+ytt comments are then used to calculate Starlark expressions to insert and concatenate this name, label, and version information into YAML nodes in multiple places like the metadata, labelSelector, and container image tag.
+
+```yaml
+#@ version
+#@ labels(with_version=True)
+#@ app_name + "-service"
+```
+
+Look at file and run in ytt playground:
+https://carvel.dev/ytt/#example:example-extract-yaml-fragments 
```

Comment:
```suggestion
[Look at this example](https://carvel.dev/ytt/#example:example-extract-yaml-fragments) and run it in the ytt playground:
```

Reply:
Does it make it less obvious when done like that that that link is for the playground and is showing the files? And that it's one of the two options?

Reply:
Yeah I debated with myself back and forth on it for a bit, I'm happy to keep it as is too